server/audit_math/sampler.py

```diff
@@ -125,20 +128,24 @@ def draw_ppeb_sample(
         for batch in batch_results
     ]
 
+    is_full_hand_tally = sample_size >= len(batch_results)
+    num_previously_sampled_batches = len(previously_sampled_batch_keys)
+
     sample: List = (
-        cast(
+        # When the sample size indicates a full hand tally, ensure that we draw all batches, minus
+        # batches already audited in previous rounds
+        sorted(list(batch_results.keys() - previously_sampled_batch_keys))
```

Comment:
I'm kind of surprised the typechecker is okay without having to cast this, but I guess that's a good thing.